pkg/cimfs/format/format.go

```diff
@@ -0,0 +1,64 @@
+//go:build windows
+// +build windows
+
+// format package maintains some basic structures to allows us to read header of a cim file. This is mostly
+// required to understand the region & objectid files associated with a particular cim. Otherwise, we don't
+// need to parse the cim format.
+package format
+
+import "github.com/Microsoft/go-winio/pkg/guid"
+
+const (
+	RegionFileName   = "region"
+	ObjectIDFileName = "objectid"
+)
+
+// Magic specifies the magic number at the beginning of a file.
+type Magic [8]uint8
+
+var MagicValue = Magic([8]uint8{'c', 'i', 'm', 'f', 'i', 'l', 'e', '0'})
+
+type Version struct {
+	Major, Minor uint32
+}
+
+var CurrentVersion = Version{3, 0}
+
+var MinSupportedVersion = Version{2, 0}
+
+type FileType uint8
+
+// RegionOffset encodes an offset to objects as index of the region file
+// containing the object and the byte offset within that file.
+type RegionOffset uint64
```

Comment:
I'm a little confused by this description. This is meant to be the offset to the objects AND the byte offset within a file?

Reply:
For each cim, we have multiple region files that store the actual contents of the data files (/objects) within cimfs. For every, object the `RegionOffset` holds the index of this region file (in 16 bits) and the offset within that region file (in 48 bits). Does that help?

Reply:
Oh, I see. Is there anywhere in our code (here or in future PRs) that we need to parse those values from a `RegionOffset`? Could we add a comment about which bits represent which thing?

Reply:
We don't need to deal with this anywhere in our code (even in the future PRs) so I didn't add any more details about using this RegionOffset.